### ISSUE TITLE
[Bug] (Datasource: Cloudwatch) Period not working at higher time ranges

### ISSUE THREAD

**Issue by Xopherus** [grafana/grafana#8962]
### Grafana Version
4.3.2

### Datasource 
Cloudwatch

### OS
Docker

### What did you do?
I created a graph with the following AWS/SQS metrics: (specifically, *NumberOfMessagesSent*, *NumberOfMessagesReceived*, *NumberOfMessagesDeleted*, *ApproximateNumberOfMessagesVisible*).

I've made a interval variable called `$period` which controls the Period parameter in the Cloudwatch metrics. It can be set to one of the following: (5m,10m,15m,30m,1h,6h,12h,1d,7d,14d,30d).

Changed time range to 30d and `$period` to 1d.

### What was the expected result?
The graph loads with 30 datapoints per metric (30 data points).

### What happened instead?
The graph loads with 1 datapoint per 30 minutes per metric (2*24*30 data points)

**Comment by mtanda** [comment]
Maybe there are bug in period calculation. Please try nightly build of today or tomorrow.

**Comment by Xopherus** [comment]
@mtanda I tried 4.4.2 this morning, no changes.

I did some additional investigation yesterday and found that period=1d DOES work for time ranges <15d. Once you go to 16d the bug appears.

I was able to dump a few of the request bodies as well:

```
// time_range=16d, period=1d
{"region":"us-west-2","action":"GetMetricStatistics","parameters":{"namespace":"AWS/SQS","metricName":"NumberOfMessagesReceived","dimensions":[{"Name":"QueueName","Value":"my-queue-prod"}],"statistics":["Sum"],"extendedStatistics":[],"startTime":1500298322,"endTime":1501594322,"period":86400}}

// time_range=15d, period=1d
{"region":"us-west-2","action":"GetMetricStatistics","parameters":{"namespace":"AWS/SQS","metricName":"NumberOfMessagesReceived","dimensions":[{"Name":"QueueName","Value":"my-queue-prod"}],"statistics":["Sum"],"extendedStatistics":[],"startTime":1500298941,"endTime":1501594941,"period":86400}}
```

**Comment by mtanda** [comment]
The fix may not included in 4.4.2 release.
Please try nightly build.

https://grafana.com/grafana/download/5.0.0-8508pre1
or next day build.

**Comment by Xopherus** [comment]
@mtanda is that equivalent to the master build on [docker hub](https://hub.docker.com/r/grafana/grafana/tags/)? it would definitely help if there was already a pre-built image?

**Comment by torkelo** [comment]
https://grafana.com/grafana/download/5.0.0-8518pre1

**Comment by Xopherus** [comment]
For reference, I had to modify https://github.com/grafana/grafana-docker to curl that particular version.

Seems like it worked! Can you identify which commit / pull request resolved this issue so the [changelog](https://github.com/grafana/grafana/blob/master/CHANGELOG.md) may be updated with this? That would be extremely helpful.

**Comment by daniellee** [comment]
Commit: https://github.com/grafana/grafana/pull/8958/commits/72251ebe2b1beb238bb7be403f4089b175f5050b

Fixed by #8958

It will be added to the changelog before we release 5.0.0.
